tools/shared.py

```diff
@@ -1167,6 +1165,9 @@ def load(self, args=[]):
           declare = re.sub(r'([\w\d]+)\s*=\s*(.+)', r'self.attrs["\1"]=\2;', args[i+1])
           exec declare
 
+      if get_llvm_target() == WASM_TARGET:
```

Comment:
This does have one functional change IIUC that you can no longer force the wasm backend by setting EMCC_WASM_BACKEND in the environment. in a hypothetical future where we have a backend that supports both asmjs and wasm, I'm not sure what `get_llvm_target` will return, but we might need to keep some kind of override.